bundles/framework/myplaces3/CategoryHandler.js

```diff
@@ -0,0 +1,700 @@
+                    },
+                    line: {
+                        cap: category.getLineCap(),
+                        corner: category.getLineCorner(),
+                        style: category.getLineStyle(),
+                        width: category.getLineWidth(),
+                        color: category.getLineColor()
+                    },
+                    area: {
+                        lineWidth: category.getAreaLineWidth(),
+                        lineCorner: category.getAreaLineCorner(),
+                        lineStyle: category.getAreaLineStyle(),
+                        lineColor: category.getAreaLineColor(),
+                        fillColor: category.getAreaFillColor(),
+                        fillStyle: category.getAreaFillStyle()
+                    }
+                };
+            form.setValues(values);
+            var content = form.getForm();
+            content.find('input[data-name=categoryname]').val(category.name);
+
+            var dialog = Oskari.clazz.create('Oskari.userinterface.component.Popup'),
+                buttons = [],
+                saveBtn = Oskari.clazz.create('Oskari.userinterface.component.Button');
+            saveBtn.setTitle(me.loc('buttons.save'));
+            saveBtn.addClass('primary');
+            saveBtn.setHandler(function () {
+                var values = form.getValues(),
+                    errors = me.validateCategoryFormValues(values);
+                if (errors.length !== 0) {
+                    me.showValidationErrorMessage(errors);
+                    return;
+                }
+                var category = me.getCategoryFromFormValues(values);
+                me.saveCategory(category);
+
+                dialog.close();
+                me.instance.sandbox.postRequestByName('EnableMapKeyboardMovementRequest');
+            });
+            var cancelBtn = Oskari.clazz.create('Oskari.userinterface.component.Button');
+            cancelBtn.setTitle(me.loc('buttons.cancel'));
+            cancelBtn.setHandler(function () {
+                dialog.close();
+                me.instance.sandbox.postRequestByName('EnableMapKeyboardMovementRequest');
+            });
+            buttons.push(cancelBtn);
+            buttons.push(saveBtn);
+            dialog.makeModal();
+            dialog.show(me.loc('categoryform.edit.title'), content, buttons);
+            dialog.moveTo('div.personaldata ul li select', 'right');
+            //bind listeners etc. for category form
+            form.start();
+
+        },
+        showValidationErrorMessage: function (errors) {
+            var dialog = Oskari.clazz.create('Oskari.userinterface.component.Popup'),
+                okBtn = Oskari.clazz.create('Oskari.userinterface.component.Button');
+            okBtn.setTitle(this.loc('buttons.ok'));
+            okBtn.addClass('primary');
+            okBtn.setHandler(function () {
+                dialog.close(true);
+            });
```

Comment:
You can do this with:

    var okBtn = dialog.createCloseButton(this.loc('buttons.ok'));